### PR TITLE
perf(aad-manifest): support scaffold aad manifest

### PR DESCRIPTION
**Pull request by SLdragon**
Add scaffold functon in aad plugin:
1. Add aad.template.json file to templates/appPackage/aad.template.json

### CODE REVIEW

File: packages/fx-core/src/plugins/resource/aad/plugin.ts [hunk 1]
```diff
@@ -587,4 +598,20 @@ export class AadAppForTeamsImpl {
 
     return requiredResourceAccessList;
   }
+
+  public async scaffold(ctx: PluginContext): Promise<AadResult> {
+    if (isAadManifestEnabled() && !isSPFxProject(ctx.projectSettings)) {
```
Comment (KennethBWSong) [inline packages/fx-core/src/plugins/resource/aad/plugin.ts:603]: I remember that only scaffold() of plugins that activated will be called. AAD plugin should not activate for spfx projects. Maybe we can remove the check? You may confirm with @jayzhang

File: packages/fx-core/src/plugins/resource/aad/plugin.ts [hunk 2]
```diff
@@ -587,4 +598,20 @@ export class AadAppForTeamsImpl {
...
+      const templatesFolder = getTemplatesFolder();
+      const appDir = `${ctx.root}/${Constants.appPackageFolder}`;
+      const solutionSettings: AzureSolutionSettings = ctx.projectSettings
+        ?.solutionSettings as AzureSolutionSettings;
+      if (isAADEnabled(solutionSettings)) {
```
Comment (KennethBWSong) [inline packages/fx-core/src/plugins/resource/aad/plugin.ts:608]: same for here

File: packages/fx-core/src/plugins/resource/aad/plugin.ts [hunk 3]
```diff
@@ -587,4 +598,20 @@ export class AadAppForTeamsImpl {
...
+        const aadManifestTemplate = `${templatesFolder}/${Constants.aadManifestTemplateFolder}/${Constants.aadManifestTemplateName}`;
+        await fs.ensureDir(appDir);
+        await fs.copy(aadManifestTemplate, `${appDir}/${Constants.aadManifestTemplateName}`);
+        solutionSettings.capabilities.push(Constants.aadCapabilityName);
```
Comment (KennethBWSong) [inline packages/fx-core/src/plugins/resource/aad/plugin.ts:612]: capability should be added by solution?

Comment (SLdragon) [reply packages/fx-core/src/plugins/resource/aad/plugin.ts]: Updated

**Review by KennethBWSong — APPROVED** [review]

**Review by KennethBWSong — APPROVED** [review]

**Pull request merged**